File: FileworxAPI/Controllers/ReceiveContactsController.cs
using FileworxObjectClassLibrary.Models;
using FileworxObjectClassLibrary.Queries;
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Mvc;

namespace FileworxAPI.Controllers
{
    [Route("api/[controller]")]
    [ApiController]
    public class ReceiveContactsController : ControllerBase
    {

        [HttpGet]
        public async Task<IActionResult> GetReceiveContacts()
        {
            try
            {
                clsContactQuery query = new clsContactQuery();
                query.QDirection = new ContactDirection[] { ContactDirection.Receive, (ContactDirection.Transmit | ContactDirection.Receive) };
                var receiveContacts = await query.RunAsync();

                return Ok(receiveContacts);
            }
            catch (Exception ex)
            {
                // Handle any exceptions that may occur during file retrieval
                return StatusCode(500, $"Internal Server Error: {ex.Message}");
            }
        }
    }
}

File: VBFileworxAPI/Controllers/ReceiveContactsController.vb
Imports System.Net
Imports System.Net.Http
Imports System.Threading.Tasks
Imports System.Web.Http
Imports FileworxObjectClassLibrary
Imports FileworxObjectClassLibrary.Models
Imports FileworxObjectClassLibrary.Queries

Namespace Controllers

    Public Class ReceiveContactsController
        Inherits ApiController

        Public Async Function GetValues() As Task(Of HttpResponseMessage)
            Try
                Dim query As New clsContactQuery()
                query.QDirection = New ContactDirection() {ContactDirection.Receive, (ContactDirection.Transmit Or ContactDirection.Receive)}
                query.Source = QuerySource.DB

                Dim receiveContacts = Await query.RunAsync()

                ' Return a 200 OK response with data (replace "receiveContacts" with your actual data)
                Return Request.CreateResponse(HttpStatusCode.OK, receiveContacts)

            Catch ex As Exception
                ' Return a 500 Internal Server Error response
                Return Request.CreateResponse(HttpStatusCode.InternalServerError, $"{ex.Message}")
            End Try
        End Function

    End Class
End Namespace